src/node_credentials.cc

```diff
@@ -33,6 +33,44 @@ bool linux_at_secure = false;
 
 namespace credentials {
 
+// Look up environment variable and allow if linux capabilities have been set
+// but not setuid. setuid is assumed if the real/effective user/group differ.
+// We also check for the off chance that the process is running as root and
+// also has capabilities set (in which case the user/group ids would be the
+// same).
+bool SafeCapGetenv(const char* key, std::string* text) {
+#if !defined(__CloudABI__) && !defined(_WIN32)
+  if (per_process::linux_at_secure && (getuid() != geteuid() ||
+      getgid() != getegid()) || getuid() == 0) {
```

Comment:
The comment here mismatches the code –  right now, this function always bails out if `getuid() == 0`, not just when capabilities are set.

Reply:
Ah indeed, I'll update this shortly. Thanks!